src/app/api/environment-variables.api.ts

```diff
@@ -0,0 +1,131 @@
+import { first, map, Observable } from "rxjs";
+import {
+    DeleteEnvVariableGQL,
+    DeleteEnvVariableMutation,
+    ExposedEnvVariableValueGQL,
+    ExposedEnvVariableValueQuery,
+    ListEnvVariablesGQL,
+    ListEnvVariablesQuery,
+    ModifyEnvVariableGQL,
+    ModifyEnvVariableMutation,
+    SaveEnvVariableGQL,
+    SaveEnvVariableMutation,
+} from "./kamu.graphql.interface";
+import { Injectable } from "@angular/core";
+import { noCacheFetchPolicy } from "../common/data.helpers";
+import { ApolloQueryResult } from "@apollo/client";
+import { DatasetOperationError } from "../common/errors";
+import { MutationResult } from "apollo-angular";
+import { DatasetApi } from "./dataset.api";
+
+@Injectable({
+    providedIn: "root",
+})
+export class EnvironmentVariablesApi {
+    constructor(
+        private listEnvVariablesGQL: ListEnvVariablesGQL,
+        private saveEnvVariableGQL: SaveEnvVariableGQL,
+        private modifyEnvVariableGQL: ModifyEnvVariableGQL,
+        private deleteEnvVariableGQL: DeleteEnvVariableGQL,
+        private exposedEnvVariableValueGQL: ExposedEnvVariableValueGQL,
+    ) {}
+
+    public listEnvironmentVariables(params: {
+        accountName: string;
+        datasetName: string;
+        page: number;
+        perPage: number;
+    }): Observable<ListEnvVariablesQuery> {
+        return this.listEnvVariablesGQL.watch({ ...params }, noCacheFetchPolicy).valueChanges.pipe(
+            first(),
+            map((result: ApolloQueryResult<ListEnvVariablesQuery>) => {
+                return result.data;
+            }),
+        );
+    }
+
+    public saveEnvironmentVariable(params: {
+        accountId: string;
+        datasetId: string;
+        key: string;
+        value: string;
+        isSecret: boolean;
+    }): Observable<SaveEnvVariableMutation> {
+        return this.saveEnvVariableGQL
+            .mutate(
+                { ...params },
+                {
+                    update: (cache) => {
+                        const datasetKeyFragment = DatasetApi.generateDatasetKeyFragment(
+                            cache.identify(DatasetApi.generateAccountKeyFragment(params.accountId)),
+                            params.datasetId,
+                        );
+                        cache.evict({
+                            id: cache.identify(datasetKeyFragment),
+                            fieldName: "envVars",
+                        });
```

Comment:
We should have some helper class to reset dataset object parts. This is complex enough to copy paste.

Reply:
Moved to helper method